### DISCUSSION TITLE
Can someone help me with this error?

### DISCUSSION THREAD

**Discussion by davidvontamar** [orbitalquark/textadept#185] [General]
I upgraded from Textadept 11.0 to Textadept 11.3. I compiled both of them on my machine with GTK3 enabled. However, once running Textadept 11.3 with my old configuration directory in place, I get the error:

> error loading module 'socket.core' from file '/usr/lib/x86_64-linux-gnu/lua/5.3/socket/core.so':
> /usr/lib/x86_64-linux-gnu/lua/5.3/socket/core.so: undefined symbol: lua_newuserdata

Each time I try to open or focus on a Lua file. In addition, I am using Linux Debian Sid and I have all Lua 5.1, 5.2, 5.3, and LuaJIT installed on my system. I don't recall how I figured it out, but I do recall that there were similar difficulties with 11.0. 11.0 worked so far without any problems until 11.3. 

Besides, there are the usual problems with the spellchecker on Debian. It refuses to work unless I do some adjustments in the source files every time I compile Textadept. I recall that these were the cross-platform issues. Another important note: The binary that's available on the website had the same “socket” error and the spellchecker there threw an error as well. Therefore, I don't think it's related to the compiling procedure on my machine.

**Comment by orbitalquark** [comment]
You should also update/recompile your modules. Textadept has its own version of Lua (5.4) and modules with shared objects, like spellcheck, file-diff, debugger, and YAML, should be using it. If LuaSocket for Lua 5.3 is being loaded, a module with an old `package.cpath` is probably causing trouble.

**Reply by davidvontamar** [reply]
Yes! I forgot to update the Lua modules in my `~/.textadept` directory.  Now it works. Thanks!
